test/ingestion-server/custom-resource/update-alb-rules.test.ts

```diff
@@ -871,134 +1210,78 @@ test('Update Listener for ClickStream SDK, with HTTP, with appIds, and have exis
         Field: 'path-pattern',
         Values: ['/collect'],
       },
-    ],
-    Priority: 7,
-  });
-
-  expect(albClientMock).toHaveReceivedNthCommandWith(7, DeleteRuleCommand, {
-    RuleArn: 'RuleArn5',
-  });
-});
-
-test('Delete all Listener rules for ClickStream SDK, with HTTPS and auth, with appIds, and with existing appId rules', async () => {
-  const event: CloudFormationCustomResourceEvent = {
-    ...basicEvent,
-    RequestType: 'Delete',
-    ResourceProperties: {
-      ...basicEvent.ResourceProperties,
-      appIds: 'nodePad1, nodePad2',
-      clickStreamSDK: 'Yes',
-      authenticationSecretArn: 'arn:aws:secretsmanager:us-east-1:11111111111:secret:test-secret',
-      protocol: 'HTTPS',
-    },
-  };
-
-  albClientMock.on(DescribeRulesCommand).resolves({
-    Rules: [
-      {
-        RuleArn: 'forwardRuleArn1',
-        IsDefault: true,
-      },
-      {
-        RuleArn: 'forwardRuleArn2',
-        IsDefault: false,
-      },
-      {
-        RuleArn: 'forwardRuleArn3',
-        IsDefault: false,
-      },
       {
-        RuleArn: 'forwardRuleArn4',
-        IsDefault: false,
+        Field: 'host-header',
+        Values: ['example.com'],
       },
     ],
+    Priority: 6,
   });
 
-  await handler(event, c);
-
-  expect(albClientMock).toHaveReceivedNthCommandWith(1, DescribeRulesCommand, {
-    ListenerArn: 'arn:aws:elasticloadbalancing:us-east-1:11111111111:listener/app/abc/026af704b2ff1dcc',
-  });
-
-  expect(albClientMock).toHaveReceivedNthCommandWith(3, DeleteRuleCommand, {
-    RuleArn: 'forwardRuleArn2',
-  });
-  expect(albClientMock).toHaveReceivedNthCommandWith(4, DeleteRuleCommand, {
-    RuleArn: 'forwardRuleArn3',
-  });
-  expect(albClientMock).toHaveReceivedNthCommandWith(5, DeleteRuleCommand, {
-    RuleArn: 'forwardRuleArn4',
+  expect(albClientMock).toHaveReceivedNthCommandWith(12, DeleteRuleCommand, {
+    RuleArn: 'RuleArn4',
   });
 });
 
-test('Update endpoint path for ClickStream SDK', async () => {
+test('Update Listener for ClickStream SDK, with HTTP, have existing appId rules, add appIds change domain, endpoint path and auth secret arn', async () => {
   const event: CloudFormationCustomResourceEvent = {
     ...basicEvent,
     RequestType: 'Update',
     ResourceProperties: {
       ...basicEvent.ResourceProperties,
-      appIds: 'notepad1,notepad2',
-      endpointPath: '/collectnew',
-      clickStreamSDK: 'YES',
-      authenticationSecretArn: 'arn:aws:secretsmanager:us-east-1:11111111111:secret:test-secret',
+      appIds: 'notepad1,notepad2,nodePad3',
+      clickStreamSDK: 'Yes',
+      authenticationSecretArn: '',
       protocol: 'HTTP',
+      authenticationSecretArn: '',
+    },
+    OldResourceProperties: {
```

Comment:
The old properties do not have `appIds`. So will the case work for updating existing rules with the application ID?

Reply:
We do not use appIds in old properties, just use appIds in new properties to compare with existing rules and update with new appIds.
In this cases, the existing rules contains nodePad3 and nodePad4, new appIds is notepad1,notepad2,nodePad3.
The notepad1,notepad2 rules will be added, and nodePad4 rule will be deleted.
The below code is checking this:
```
  expect(albClientMock).toHaveReceivedNthCommandWith(9, DeleteRuleCommand, {
    RuleArn: 'RuleArn5',
  });
```